spec/actors/iiif_print/actors/iiif_print_upload_actor_spec.rb

```diff
@@ -34,130 +41,86 @@
             work,
-            ["/app/samvera/hyrax-webapp/.internal_test_app/tmp/uploads/hyrax/uploaded_file/file/1/minimal-2-page.pdf"],
-            "spaceballs@example.com",
-            "admin_set/default"
+            pdf_paths,
+            my_user,
+            "admin_set/default",
+            0
           )
           expect(middleware.public_send(mode, mode_env)).to be true
         end
       end
       context 'when work has no pdf file' do
         let(:mode_env) { no_pdf_env }
-        it 'does not queue IiifPrint::CreatePagesJob' do
-          expect(IiifPrint::CreatePagesJob).not_to receive(:perform_later)
+        it 'does not queue job' do
+          expect(IiifPrint::Jobs::ChildWorksFromPdfJob).not_to receive(:perform_later)
           expect(middleware.public_send(mode, mode_env)).to be true
         end
       end
     end
 
     describe ':update' do
       let(:mode) { :update }
-      before do
-        allow(work).to receive(:respond_to?).and_call_original
-        allow(work).to receive(:respond_to?).with(:split_pdf).and_return true
-      end
-      context 'works is updated with no additional uploads' do
+      context 'work is updated with no additional uploads' do
         let(:mode_env) { edit_env }
-        it 'queues a IiifPrint::CreatePagesJob' do
-          expect(IiifPrint::CreatePagesJob).not_to receive(:perform_later)
+        it 'does not queue job' do
+          expect(IiifPrint::Jobs::ChildWorksFromPdfJob).not_to receive(:perform_later)
+          expect(middleware.public_send(mode, mode_env)).to be true
+        end
+      end
+      context 'work is updated with an additional PDF' do
+        let(:mode_env) { with_pdf_env }
+        it 'queues IiifPrint::Jobs::ChildWorksFromPdfJob' do
+          expect(IiifPrint::Jobs::ChildWorksFromPdfJob).to receive(:perform_later).with(
+            work,
+            pdf_paths,
+            my_user,
+            "admin_set/default",
+            0
+          )
           expect(middleware.public_send(mode, mode_env)).to be true
         end
       end
     end
   end
 
-  context 'when work model does not IiifPrintBehavior' do
+  context 'when work model does not use IiifPrint' do
+    let(:work) { work_without_config }
     describe ':create' do
       let(:mode) { :create }
-      before do
-        allow(work).to receive(:respond_to?).and_call_original
-        allow(work).to receive(:respond_to?).with(:split_pdf).and_return false
-      end
       context 'when work has a pdf file' do
         let(:mode_env) { with_pdf_env }
-        it 'queues a IiifPrint::CreatePagesJob' do
-          expect(IiifPrint::CreatePagesJob).not_to receive(:perform_later)
+        it 'does not queue IiifPrint::Jobs::ChildWorksFromPdfJob' do
+          expect(IiifPrint::Jobs::ChildWorksFromPdfJob).not_to receive(:perform_later)
           expect(middleware.public_send(mode, mode_env)).to be true
         end
       end
       context 'when work has no pdf file' do
```

Comment:
do you think we still need this context for the `when work model does not use IiifPrint` context?

Reply:
since work models do not always have the iiif_print configuration (that config existing for the model is what determines whether or not the work does this extra behavior), I thought it should have a test for it. Why would you think it shouldn't be there?